sorc/orog_mask_tools.fd/orog_gsl.fd/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+set(fortran_src
+    module_gsl_oro_data_lg_scale.f90
+    module_gsl_oro_data_lg_scale.sar.f90
+    module_gsl_oro_data_sm_scale.f90
+    module_gsl_oro_data_sm_scale.sar.f90
+    gsl_oro_data.f90)
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -r8 -convert big_endian -assume byterecl")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fconvert=big-endian -fno-range-check")
+  if(CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz")
+  endif()
+endif()
+
+set(exe_name orog_gsl)
+add_executable(${exe_name} ${fortran_src})
+target_link_libraries(
+  ${exe_name}
```

Comment:
Your program is self-contained.  So there is no need to map in the bacio, w3nco, ip or sp libraries.  And if you are not using threads, you can remove the reference to OpenMP_Fortran.

Reply:
I modified CMakeLists.txt according to your suggestions.